test/new-e2e/utils/e2e/client/agent.go

```diff
@@ -41,16 +41,24 @@ func (agent *Agent) initService(t *testing.T, data *agent.ClientData) error {
 	return err
 }
 
-func (agent *Agent) Version() string {
-	return agent.vmClient.Execute(agent.GetCommand("version"))
-}
-
 func (agent *Agent) GetCommand(parameters string) string {
 	return agent.os.GetRunAgentCmd(parameters)
 }
 
-func (agent *Agent) Config() string {
-	return agent.vmClient.Execute(agent.GetCommand("config"))
+func (agent *Agent) executeCommand(command string, commandArgs ...AgentArgsOption) string {
+	args, err := newAgentArgs(commandArgs...)
+	if err != nil {
+		return ""
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Not a big fan of hiding errors, I'd rather use the `ExecuteWithError` for `WaitForReady` reusing `newStatus`, something like

```
func (agent *Agent) WaitForReady() err {
   err := loopOnError({
     statusOutput, err := agent.ExecuteWithError("status")
     if err {
       return err
     }
     status := newStatus(statusOutput)
     if !status.isReady() {
       return new error
     }
     return nil
  })
  return err
}

Reply:
I'm not a fan either, and I realised I forgot to talk about it in the PR description.

My thoughts was that we should return (string, error) for all the Agent commands (I already have silent error; e.g. got an empty string from a failing command) and use `ExecuteWithError` within `executeCommand()`.

For the WaitForReady() example you gave, do you want me to apply it on this PR and re introduce `requireNoError` in `Execute` ?

Reply:
Yes please !